### PR TITLE
Adds support to read more than 200 groups for a user

### PR DESCRIPTION
**Pull request by ntroncos**

### CODE REVIEW

File: okta/users.go [hunk 1]
```diff
@@ -261,19 +263,24 @@ type UserListFilterOptions struct {
...
+			return nil, err
+		}
+		resp, err := s.client.Do(req, &groups)
+		if err != nil {
+			return resp, err
+		}
+		user.Groups = append(user.Groups, groups...)
+		if len(groups) < oktaPaginateSize {
+			return resp, err
+		}
```
Comment (vishalnayak) [inline okta/users.go:281]: Will the len(groups) never be greater than 200?

**Comment by ntroncos** [comment]
Not according to OKTA documentation, but I see where you are coming from.

Let me run some tests with the OKTA API to see if I can make it size agnostic.

**Comment by calvn** [comment]
Wouldn’t this make an additional request (that would return an empty group array) if the total group size is a number that’s divisible by 200?

**Comment by ntroncos** [comment]
It will. There is no way around that since okta gives you no indication about the number of groups a user has.

**Comment by calvn** [comment]
According to the docs, the response should return a Link header that is a list of comma-separated links. It might be addressable by doing some header parsing and looking for the existence of a link with `rel="next"`. If that can be used for groups listing, your next request could simply be that URL instead of building it manually within the for loop. You wouldn't have to worry about the max group size returned either.

https://developer.okta.com/docs/api/getting_started/design_principles#link-header

**Comment by ntroncos** [comment]
I see the link in the headers. I will try to work some magic today and get it working.

**Comment by chrismalek** [comment]
Yes there should be a "nextPage" header with a URL that is returned and is the "proper" way to do pagination. I am already doing this pagination header parsing in the groups api.  I remembered I had some trouble getting the header to parse there.  

Example header:

`< Link: <https://someorg.okta.com/api/v1/users/00uwtat3gdULXDYKQOYA/groups?after=00g126cocigQ4Izdr1e8&limit=5>; rel="next"`

**Comment by chrismalek** [comment]
I just pushed a change that should pull all user groups. See https://github.com/chrismalek/oktasdk-go/commit/d0c464c6e7d0c3407d2ec2da7a5818536e600515


Let me know if that does not address it for you.

**Comment by ntroncos** [comment]
@chrismalek 
Look like it's working as advertised. 

`13:58:03.408188 [DEBUG] auth/okta: Groups fetched from Okta: num_groups=368`

This is good for me.

**Pull request closed without merging**